src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -125,91 +163,190 @@ class FlightTaskAutoFollowTarget : public FlightTask
+	 * Calculate the tracked target orientation and overwrite the tracked target orientation if necessary
 	 *
-	 * @param deltatime [s] prediction horizon
-	 * @return Future prediction of target position
+	 * Note : Filtered target velocity is generated via 2nd order reference model filter can have overshooting behaviors
+	 * when target stops it's motion. This can generate a target velocity output that is opposite direction to where target was heading originally.
+	 * To check if the filtered velocity is staying true to target's actual motion, unfiltered velocity needs to be taken into account. Since during
+	 * overshoot, the unfiltered velocity stays close to 0 (indicating target already stopped), therefore not triggering a target orientation setting.
+	 *
+	 * @param current_target_orientation  Tracked target orientation value that will be over-written with new orientation
+	 * @param target_velocity Filtered Target velocity from which we will calculate target orientation
+	 * @param target_velocity_unfiltered Unfiltered Target velocity that aids in verifying if filtered velocity is accurate
+	 */
+	void update_target_orientation(float &current_target_orientation, const Vector2f &target_velocity, const Vector2f &target_velocity_unfiltered);
+
+	/**
+	 * Updates the orbit angle setpoint, taking into account the maximal orbit tangential speed
+	 *
+	 * Returns the orbit angle setpoint, taking into account the maximal orbit tangential speed.
+	 * While setting the orbital velocity setpoint vector for the according position setpoint
+	 *
+	 * @param target_orientation Tracked target orientation
+	 * @param previous_orbit_angle_setpoint Previous orbit angle setpoint
+	 *
+	 * @return Angle [rad] Next feasible orbit angle setpoint
 	 */
-	matrix::Vector3f predict_future_pos_ned_est(float deltatime, const matrix::Vector3f &pos_ned_est,
-			const matrix::Vector3f &vel_ned_est,
-			const matrix::Vector3f &acc_ned_est) const;
+	float update_orbit_angle_trajectory(const float target_orientation, const float previous_orbit_angle_setpoint);
 
+	/**
+	 * Returns the orbit angle setpoint, taking into account the maximal orbit tangential speed.
+	 *
+	 * @param orbit_angle_setpoint Orbit angle setpoint
+	 *
+	 * @return 2D Velocity Vector [m/s, m/s] of current orbit position setpoint
+	*/
+	Vector2f get_orbit_tangential_velocity(const float orbit_angle_setpoint) const;
+
+	/**
+	 * Calculates desired drone position taking into account orbit angle and the follow target altitude mode
+	 *
+	 * @param target_position Tracked target position Vector3f reference
+	 * @param orbit_angle_setpoint Current orbit angle setpoint around the target
+	 *
+	 * @return Position [m,m,m] Final position setpoint for the drone
+	 */
+	Vector3f calculate_desired_drone_position(const Vector3f &target_position, const float orbit_angle_setpoint) const;
+
+	/**
+	 * Calculate the gimbal height offset to the target to calculate the pitch angle command
```

Comment:
```suggestion
	 * Calculate the gimbal height offset to the target
```
it is not necessary that the return be used for pitch considerations, just talk about what *this method does.